lib/iris/analysis/__init__.py

```diff
@@ -1191,20 +1225,27 @@ def _weighted_percentile(data, axis, weights, percent, returned=False,
         return result
 
 
-def _count(array, function, axis, **kwargs):
-    if not callable(function):
-        raise ValueError('function must be a callable. Got %s.'
-                         % type(function))
-    return ma.sum(function(array), axis=axis, **kwargs)
+@_build_dask_mdtol_function
+def _lazy_count(array, **kwargs):
+    array = iris._lazy_data.as_lazy_data(array)
+    try:
+        func = kwargs.pop('function')
+    except KeyError:
+        raise KeyError('no selection function supplied.')
```

Comment:
I'm +0 on this extra exception. Without it (assuming you do ``func = kwargs.pop('function', None)`` we get a TypeError, which is good enough for me (I accept that KeyError give you even more exception fidelity, but my argument is that the fidelity isn't worth the extra lines).